### PR TITLE
Change `pre-release.yml` for multi-platform builds

### PR DESCRIPTION
**Pull request by TheGupta2012**
## Summary of changes
* Added platform specific builds in `pre-release.yaml`
* Updated the `pre_build.sh` script to detect whether we are in pre-release, if yes, then change the version of the toml file to dev version.
*  Updated the `CIBW_BEFORE_BUILD` command to include the project name as an argument.

### CODE REVIEW

**Review by ryanhill1 — CHANGES_REQUESTED** [review]
With these updates, we no longer need the `create_dev_build.sh` script, right?

**Comment by TheGupta2012** [comment]
Yea, idts that the script would be needed.

The only thing which is unclear to me is how the pre-release version numbers would be tracked. For instance, say we have published `v0.1.0-a1`. Once we trigger the pre-release flow, how will it know that the next version to be published is `v0.1.0-a2` if the toml version stays at `v0.1.0-alpha`?

File: bin/cibw/pre_build.sh
[the remark this answers is not in the document]
Comment (TheGupta2012) [reply bin/cibw/pre_build.sh:1]: Yep

**Comment by ryanhill1** [comment]
@TheGupta2012 It does so using [qbraid_core.system.versions.get_latest_package_version()](https://github.com/qBraid/qbraid-core/blob/v0.1.29/qbraid_core/system/versions.py#L167) with the option `prerelease=True`. This function makes a `GET` request to PyPI, and gets back a list of all of the releases for the given package, and is used as a helper within [qbraid_core.system.versions.get_prerelease_version()](https://github.com/qBraid/qbraid-core/blob/v0.1.29/qbraid_core/system/versions.py#L332). And then if the `pyproject.toml` version ends in `-alpha`, we get that latest "alpha" release matching that version and bump it by one. Same thing for "beta" and "rc". Or if no pre-releases for that marker exist yet, we set it to 0.

File: bin/cibw/pre_build.sh
[the remark this answers is not in the document]
Comment (TheGupta2012) [reply bin/cibw/pre_build.sh:1]: Although we can make it optional, for consistency in the workflows I've provided it in both `main` and `release`

**Review by ryanhill1 — APPROVED** [review]

**Comment by TheGupta2012** [comment]
@ryanhill1 have also added the change for `sdist` version bump

**Pull request merged**